File: src/main/java/rars/venus/settings/editor/views/PickerCardView.java
package rars.venus.settings.editor.views;

import org.jetbrains.annotations.NotNull;

import javax.swing.*;
import java.awt.*;

public final class PickerCardView extends JPanel {
    public final @NotNull BaseStyleView baseStyleView;
    public final @NotNull PresetsView presetsView;
    public final @NotNull SyntaxStyleView syntaxStyleView;
    public final @NotNull FontSettingsView fontSettingsView;
    public final @NotNull OtherSettingsView otherSettingsView;
    private final @NotNull JPanel upperPanel;
    private final @NotNull CardLayout upperLayout;
    
    private static final @NotNull  String EMPTY = "empty",
            FONT = "font",
            PRESETS = "presets",
            BASE = "base",
            SYNTAX = "syntax",
            OTHER = "other";

    public PickerCardView() {
        super();
        setLayout(new BorderLayout());
        upperLayout = new CardLayout();
        upperPanel = new JPanel(upperLayout);

        setPreferredSize(new Dimension(450, 450));
        setMinimumSize(new Dimension(50, 50));

        final var emptyPanel = new JPanel();
        baseStyleView = new BaseStyleView();
        presetsView = new PresetsView();
        syntaxStyleView = new SyntaxStyleView();
        fontSettingsView = new FontSettingsView();
        otherSettingsView = new OtherSettingsView();

        upperPanel.add(emptyPanel, EMPTY);
        upperPanel.add(fontSettingsView, FONT);
        upperPanel.add(presetsView, PRESETS);
        upperPanel.add(baseStyleView, BASE);
        upperPanel.add(syntaxStyleView, SYNTAX);
        upperPanel.add(otherSettingsView, OTHER);

        this.add(upperPanel, BorderLayout.NORTH);
    }

    public void showBaseStyleView() {
        upperLayout.show(upperPanel, "base");
    }

    public void showFontView() {
        upperLayout.show(upperPanel, "font");
    }

    public void showSyntaxStyleView() {
        upperLayout.show(upperPanel, "syntax");
    }

    public void showEmpty() {
        upperLayout.show(upperPanel, "empty");
    }

    public void showOtherSettings() {
        upperLayout.show(upperPanel, "other");
    }
    
    public void showPresets() {
        upperLayout.show(upperPanel, "presets");
    }
}

File: src/main/java/rars/venus/settings/editor/views/PickerCardView.kt
package rars.venus.settings.editor.views

import rars.settings.SettingsTheme
import rars.venus.util.BorderLayout
import rars.venus.util.CardLayout
import rars.venus.util.JPanel
import java.awt.BorderLayout
import java.awt.CardLayout
import java.awt.Dimension
import javax.swing.JPanel

class PickerCardView(
    initialTheme: SettingsTheme,
    initialCaretBlinkRate: Int,
    initialEditorTabSize: Int
) : JPanel() {

    val baseStyleView = BaseStyleView(initialTheme)
    val presetsView = PresetsView()
    val syntaxStyleView = SyntaxStyleView()
    val fontSettingsView = FontSettingsView()
    val otherSettingsView =
        OtherSettingsView(initialCaretBlinkRate, initialEditorTabSize)

    private val upperLayout: CardLayout
    private val upperPanel = JPanel {
        CardLayout {
            this[EMPTY] = JPanel()
            this[FONT] = fontSettingsView
            this[PRESETS] = presetsView
            this[BASE] = baseStyleView
            this[SYNTAX] = syntaxStyleView
            this[OTHER] = otherSettingsView
        }.also { upperLayout = it }
    }


    init {
        preferredSize = Dimension(450, 450)
        minimumSize = Dimension(50, 50)

        BorderLayout {
            this[BorderLayout.NORTH] = upperPanel
        }
    }

    fun showBaseStyleView() {
        upperLayout.show(upperPanel, BASE)
    }

    fun showFontView() {
        upperLayout.show(upperPanel, FONT)
    }

    fun showSyntaxStyleView() {
        upperLayout.show(upperPanel, SYNTAX)
    }

    fun showEmpty() {
        upperLayout.show(upperPanel, EMPTY)
    }

    fun showOtherSettings() {
        upperLayout.show(upperPanel, OTHER)
    }

    fun showPresets() {
        upperLayout.show(upperPanel, PRESETS)
    }

    companion object {
        private const val EMPTY = "empty"
        private const val FONT = "font"
        private const val PRESETS = "presets"
        private const val BASE = "base"
        private const val SYNTAX = "syntax"
        private const val OTHER = "other"
    }
}
